app/admin/rest.py

```diff
@@ -0,0 +1,103 @@
+from flask import Blueprint, abort, jsonify, request
+from sqlalchemy.exc import NoResultFound
+
+from app.errors import register_errors
+from app.models import (
+    ApiKey,
+    Complaint,
+    EmailBranding,
+    InboundNumber,
+    InboundSms,
+    Job,
+    LetterBranding,
+    Notification,
+    Organisation,
+    ProviderDetails,
+    Service,
+    ServiceCallbackApi,
+    ServiceContactList,
+    ServiceDataRetention,
+    ServiceEmailReplyTo,
+    ServiceInboundApi,
+    ServiceSmsSender,
+    Template,
+    TemplateFolder,
+    User,
+)
+
+admin_blueprint = Blueprint("admin", __name__)
+register_errors(admin_blueprint)
+
+FIND_BY_UUID_MODELS = {
+    "organisation": Organisation,
+    "service": Service,
+    "template": Template,
+    "notification": Notification,
+    "email_branding": EmailBranding,
+    "letter_branding": LetterBranding,
+    "user": User,
+    "provider": ProviderDetails,
+    "reply_to_email": ServiceEmailReplyTo,
+    "job": Job,
+    "service_contact_list": ServiceContactList,
+    "service_data_retention": ServiceDataRetention,
+    "service_sms_sender": ServiceSmsSender,
+    "inbound_number": InboundNumber,
+    "api_key": ApiKey,
+    "template_folder": TemplateFolder,
+    "service_inbound_api": ServiceInboundApi,
+    "service_callback_api": ServiceCallbackApi,
+    "complaint": Complaint,
+    "inbound_sms": InboundSms,
+}
+
+# The `context` here is actually information required by the admin app to build the redirect URL. This does mean
+# that this endpoint and the admin search page are fairly closely coupled. We could serialize the entire object
+# back, which might be more consistent with how other endpoints return representations of objects, but for many
+# objects we don't need any information at all, and our serialization of objects is already inconsistently between
+# `instance.serialize` and `some_schema.dump(instance)`. And not all of the serialized representations of objects
+# expose the field we need anyway (which is, as of writing, invariably the related service id).
+FIND_BY_UUID_EXTRA_CONTEXT = {
+    "template": {"service_id"},
+    "notification": {"service_id"},
+    "reply_to_email": {"service_id"},
+    "job": {"service_id"},
+    "service_contact_list": {"service_id"},
+    "service_data_retention": {"service_id"},
+    "service_sms_sender": {"service_id"},
+    "api_key": {"service_id"},
+    "template_folder": {"service_id"},
+    "service_inbound_api": {"service_id"},
+    "service_callback_api": {"service_id"},
+    "inbound_sms": {"service_id"},
+}
+
+
+@admin_blueprint.route("/find-by-uuid", methods=["POST"])
+def find_by_uuid():
+    """Provides a simple interface for looking whether a UUID references any common DB objects"""
+    uuid_ = request.json["uuid"]
+
+    instance = None
+    for entity_name, model in FIND_BY_UUID_MODELS.items():  # noqa: B007 (entity name not used in loop)
```

Comment:
im not a fan of us relying on the last value of an iterator after we've broken out of the loop - that's quite surprising control logic for me, and i easily missed the importance of using `.items` rather than `values`.

we could refactor this into a function instead? for example:

```py
def find_model_by_uuid(uuid_):
    for entity_name, model in FIND_BY_UUID_MODELS.items():
        # this is a nice alternative to except: pass even if we don't refactor the whole function
        with suppress(NoResultFound): 
            if instance := model.query.get(uuid_):
                return {'entity_name': entity_name, 'instance': instance}
    raise NoResultFound
```

Reply:
Yeah this is probably nicer, will change 👍